### PR TITLE
Custom type concept

### PR DESCRIPTION
**Pull request by lpil**

### CODE REVIEW

File: concepts/custom-types/about.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+Creating a value for a specific case can be done by referring to its name if it contains no additional data (`Spring`), or by calling it as a function if it does (`SomeInt(2)`).
+
+```gleam
+let spring = Spring
+let integerTwo = SomeInt(2)
+```
+
+Custom types, along with everything in Gleam, have _structural equality_, which means that two values of the same variant and with the same data are equivalent.
```
Comment (giacomocavalieri) [inline concepts/custom-types/about.md:31]: Maybe to be extra clear I'd show an example:
```gleam
Spring == Spring // -> True
Spring == Autumn // -> False
SomeInt(2) == SomeInt(2) // -> True
SomeInt(2) == SomeFloat(2.0) // -> False

```

File: exercises/concept/valentines-day/.docs/hints.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Hints
+
+## 1. Define the approval
+
+- Custom types can be defined using the `pub type Name { Variant1 }` syntax.
+
+## 4. Define the activity
+
+- Custom types can be defined containing data the `pub type Name { Variant1(SomeType) }` syntax.
```
Comment (giacomocavalieri) [inline exercises/concept/valentines-day/.docs/hints.md:9]:
```suggestion
- Custom types containing data can be defined using the `pub type Name { Variant1(SomeType) }` syntax.
```

File: exercises/concept/valentines-day/.meta/design.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- Know how to define a custom type variant without contained data.
+- Know how to define a custom type variant with contained data.
+- Know how to pattern match on custom types.
+
+## Out of scope
+
+- Labelled fields.
+- Record updates.
+- Record accessors.
+- Splat patterns.
```
Comment (giacomocavalieri) [inline exercises/concept/valentines-day/.meta/design.md:15]: Out of curiosity, what's a splat pattern? I don't remember reading about it in the language tour

Comment (lpil) [reply exercises/concept/valentines-day/.meta/design.md:15]: That should say spread, oops.

Comment (giacomocavalieri) [reply exercises/concept/valentines-day/.meta/design.md:15]: Aah now it makes sense! This won't stop me from calling it splat from now on, it's pretty funny 😂

Comment (lpil) [reply exercises/concept/valentines-day/.meta/design.md]: It's a Ruby term!

File: concepts/custom-types/about.md (hunk 1 above)
Comment (lpil) [reply concepts/custom-types/about.md:31]: Nice!

**Pull request merged**